platform/feature-api/api/pod-api/src/api.ts

```diff
@@ -199,6 +182,31 @@ export interface PolyLifecycle {
     startFeature(id: string, background: boolean): Promise<void>;
 }
 
+/**
+ * `Endpoint` is the API features communicate with in order to perform fetch requests
+ */
+export interface Endpoint {
+    /**
+     * Perform a http post request via the endpoint in the pod
+     * @param body the necessary content of the call
+     * @returns an empty response or throws an error if something goes wrong
+     * @throws if an unsupported request goes through, if an endpoint is not reached or if a user denies a request
+     */
+    send(
```

Comment:
OK, I'm looking at the usages of this endpoint, and one of the things that's missing is checking that payload matches the contentType that's declared. That delegates that check to the client, and it's possibly OK as a first approximation, but additionally the problem is that we're using 4 arguments, mostly untyped (strings), all across the call hierarchy, sometimes calling them by different names. This again, calls for the creation of a class that makes that a bit more typesafe, and that can be translated across language boundaries. At any rate, this is valid as a first approximation, but I would like to suggest a refinement for the next iteration, if possible. At least, "payload" and "contentType" should be in a single object, I guess.